src/main/java/seedu/address/model/transaction/Transaction.java

```diff
@@ -63,9 +63,23 @@ public int hashCode() {
         return Objects.hash(name, amount, category, dateTime, location, type);
     }
 
+    /**
+     * Returns true if both transactions have the same name and dateTime.
+     * This defines a weaker notion of equality between two transactions.
+     */
+    public boolean isSameTransaction(Transaction otherTransaction) {
+        if (otherTransaction == this) {
+            return true;
+        }
+
+        return otherTransaction != null
+                && otherTransaction.getName().equals(getName())
+                && otherTransaction.getDateTime().equals(getDateTime());
+    }
+
```

Comment:
Is this required? I initially did not add it in because we agreed that transactions did not need to have this level of check

Reply:
Removed it. Thanks for catching this!